src/Aspire.Hosting/ApplicationModel/EndpointAnnotation.cs

```diff
@@ -81,15 +64,30 @@ public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, strin
     /// <summary>
     /// Desired port for the service
     /// </summary>
-    public int? Port { get; set; }
+    public int? Port {
+        // For proxy-less Endpoints the client port and target port should be the same.
+        // Note that this is just a "sensible default"--the consumer of the EndpointAnnotation is free
+        // to change Port and TargetPort after the annotation is created, but if the final values are inconsistent,
+        // the associated resource may fail to run.
+        // It also depends on what the EndpointAnnotation is applied to.
+        // In the Container case the TargetPort is the port that the process listens on inside the container,
+        //  and the Port is the host interface port, so it is fine for them to be different.
+        get => _port ?? (IsProxied ? null : _targetPort);
```

Comment:
I am not sure about this change... this is going to result in pretty weird behavior for a **settable** property:

```csharp
var e = new EndpointAnnotation(ProtocolType.Tcp, port: 1234, targetPort: null, isProxied: false);
Console.WriteLine(e.TargetPort); // prints 1234
e.TargetPort = 4567;
Console.WriteLine(e.TargetPort); // prints 4567
e.TargetPort = null;
Console.WriteLine(e.TargetPort); // prints 1234
e.IsProxied = true;
Console.WriteLine(e.TargetPort); // prints null
```

Reply:
I was thinking the same thing. This is basically a way to get torn state. Maybe we need a gross internal method that used in the WithEndpoint callback...

Reply:
Internal method was my first thought, but it felt a bit dirty to have logic outside of `EndpointAnnotation`. And it opens the possibility that some new scenario creates an `EndpointAnnotation` and forgets to call it, leading to inconsistent fallback behavior.

An alternative is to add `_portSettingCalled`/`_targetPortSettingCalled` fields to disable the fallback if the setter was ever called.

Let me know which way you prefer, I can do either.

Reply:
I made a change along those lines. It's a bit more convoluted than I would have liked, but maybe still better than an internal method called from the outside. I also extended the tests to cover setting to null. PTAL.